File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        dic=defaultdict(int)
        for element in s:
            dic[element]+=1
        for i in t:
            dic[i]-=1
            if dic[i] < 0:
                return False
        return True

File: Data Structures & Algorithms/is-palindrome/submission-0.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        // use isalnum to skip the non-alphanumeric characters element
        // use tolower to change the highercase to lowercase character
        int l=0;        
        int r=s.size();
        while(l<r){
            while(l<r && !isalnum(s[l])) l++;
            while(l<r && !isalnum(s[r])) r--;
            if(tolower(s[l])!=tolower(s[r])) return false;
            l++;
            r--;
        }
        return true;
    }
};
